processing/src/main/java/io/druid/query/extraction/AbstractDimRenameFn.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright (C) 2012, 2013, 2014  Metamarkets Group Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package io.druid.query.extraction;
+
+import com.google.common.base.Function;
+
+import java.util.Map;
+
+/**
+ *
+ */
+public abstract class AbstractDimRenameFn implements DimExtractionFn
```

Comment:
The only thing this seems to do beyond the `Function` interface is check for null and return the value instead, which, I think is semantics that the DimExtractionFn should implement.

That is, I don't really see the point of this as an abstract class, just make it an interface, each of the points that are using this is already generating a `Function` to pass in here, so it won't actually be much extra work for them to just implement the interface.